File: 7_kyu/Get_the_Middle_Character.py
# Returns the characters in the middle of s.
# Returns 1 character if length of s is odd,
# else returns the 2 characters in the middle of the string s.
def get_middle(s: str) -> str:
    i = (len(s)-1) // 2
    j = (len(s)+2) // 2
    return s[i:j]

File: 7_kyu/Get_the_Middle_Character.rs
// Returns the middle character of the strings.
// Returns 1 or 2 character length string if s length is odd or even.
fn get_middle(s: &str) -> &str {
    // Convert string of UTF-8 bytes into an array of characters.
    let v: Vec<char> = s.chars().collect();
    let n = v.len();
    if n < 3 {
        return s;
    }
    // Calculate the byte positions in the UTF-8 string
    let prefix: String = v[0..(n - 1) / 2].iter().collect();
    let middle: String = v[(n - 1) / 2..(n + 2) / 2].iter().collect();
    // Return the middle part of the UTF-8 string, could be 1 byte up to 8 bytes.
    &s[prefix.len()..prefix.len() + middle.len()]
}
